### ISSUE TITLE
E13999 - Bluetracs (Account listed under this account E1330 - Briar Pipe)

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1143]
In GitLab by @kdjstudios on Sep 25, 2018, 16:57

**Submitted by:** "Kimberly Gagner" <kimberly.gagner@answernet.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2018-09-25-46192/conversation

**Server:** Internal

**Client/Site:** Billerica

**Account:** 123-E13999

**Issue:** 

I noticed today when putting together a list that Briar Pipes is set up in SAB as resource code E1330 and the account is set up in Pinnacle as E1330 and also E2330.  However, in Pinnacle the messages are taken in account 2330 so when it comes time to bill for talk time minutes it will not bill if E2330 is not listed as the resource code.  I attempted to go add a “new account” to Blue Tracs as Briar Pipe E2330 and I kept getting an error message.  Is this something that needs to be done in the background? 

 

Could you possibly get this done as soon as possible before my billing closes?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 25, 2018, 16:58

I wrote:

> Would you please send a screen shot of the error you are receiving. Also when are you receiving this error? Is it when trying to save the account; or trying to add activities, fees, etc to the account?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 25, 2018, 17:03

Kimberly wrote:

> "We're sorry, but something went wrong"   

I wrote:

> Could you please also answer the other questions as the support team will need to know in order to better assist. 
> 
> Also when are you receiving this error? Is it when trying to save the account; or trying to add activities, fees, etc to the account?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 25, 2018, 17:05

@smasih @pchaudhary @amishra @tim.traylor 

It looks like we received a few error emails just before this ticket was submitted. They are all titled "[SA Billing Error Report] invoice_line_items#destroy (Mongoid::Errors::DocumentNotFound) "\nProblem:\n  Document(s) not ..."

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 25, 2018, 17:20

Kimberly wrote:

> It happens no matter what you try to do because the account already exists. 

I wrote:

> I must have misread your original ticket as I thought you stated you were creating a new account. Could you please clarify what you are trying to accomplish. I will try my best to recap what I am able to see in SAB below in hopes of better understanding your request.
> 
> I see the following accounts and customers:
> 
> Account 123-E2330 is under customer Briar Pipe/Harbor Properties.

> Account 123-E1330 is under customer BlueTracs and is a child account to Account 123-E13999.

> Account 123-E13999 is under customer BlueTracs and is a parent account to multiple accounts. 
> 
> Are you wanting to move the Account 123-E2330 to be under the customer BlueTracs and have Account 123-E13999 be its parent account?

Kimberly wrote:

> E2330 if you search it in SAB it is listed as an account that is most likely why that error is coming up.  The account should be listed in E13999 under the list of accounts and it is in that list as E1330 so it is not billing correctly.  It will not allow me to add it because it recognizes it already in SAB. 
> 
> I hope this explains it better.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 25, 2018, 17:29

Kimberly wrote:

> When I was in E13999 gathering a list of the accounts still active in this account I noticed that Briar Pipe is listed as E1330 under E13999.  However, the account is set up in Pinnacle (our platform) as E2330 so when agents take messages they will bill to E2330 and not E1330 so I need E1330 to be E2330 in E13999 or the account E2330 should be moved to E13999 and set up the same way as E1330 and E1330 is not being used in Pinnacle so it can just be replaced with E2330.   

I wrote:

> Thanks I think I have a grasp at what you are attempting to do. We have informed the support team and will update you accordingly. 
> 
> I do want to mention there might be an easy solution: Simply, you should only need to update your accounts in Pinnacle because that is where they are setup incorrectly. You should not be updating anything in SAB. I would suggest you should just need to update account number E2330 to be account number E1330 in Pinnacle. This should then start tracking your usage under the correct account number in Pinnacle and then SA Billing will be able to bill accordingly. Does that make sense? NOTE: I would presume that Pinnacle will not change historical data, so you would need to manually billing for this last month's usage, but the next billing cycle will have the correct usage associated with the correct account number.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 26, 2018, 05:03

Hi [Kyle] ( @kdjstudios ),  

We have read the above conversation & want to provide you with some of the key points: 

* First, we want to ensure that what we understand what Kimberly is trying to explain is correct or not: 
  * He wants to update an account number `123-E1330` to `123-E2330`   
   **OR**   
  * He wants to set up a new account having the account number `123-E2330` under the `123-E13999` with the same things as `123-E1330`.

* So, if that is the case then the best way is to update the account number of `123-E1330` but while doing so he will have gone through the warning message to change the account number because `123-E2330` already exists under the scope of the site.  
  ![image](/uploads/b462e593bad57eb243927b9abba51332/image.png)

* For that, he must have to change the account number of `123-E2330` into something else or he can terminate the account if it's not in use, which changes the account number from `123-E2330` to `T-123-E2330`.   
* Now he can easily change the account number of `123-E1330` to `123-E2330`.

Hope this works.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 26, 2018, 05:03

added 1h of time spent at 2018-09-26

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 26, 2018, 05:03

assigned to @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 26, 2018, 10:28

@pchaudhary 

Thank you for the update on how we can accomplish her request. However before we can have Kimberly do anything we need to resolve the error Kimberly is getting. As stated in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1145#note_27573 and then it looks like the email error I found is in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1145#note_27574

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 28, 2018, 06:14

Hi [Kyle]  ( @kdjstudios ),   

We weren't able to replicate the issue in our end. Please, we request Kimberly to try again.

We also would like to mention that the error you are pointing out is for a different reason & for the different site also. There is nothing related to this issue on that error mail.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 28, 2018, 06:14

added 30m of time spent at 2018-09-28

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 28, 2018, 09:01

I wrote:

> The support team was able to look into this and has so far been unable to reproduce the error you received. They are requesting that you try again and take a screen shot of the error. If you can also include the exact steps you are taking this should be helpful.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Oct 9, 2018, 04:13

@kdjstudios 

Waiting for the response from Kimberly end.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 9, 2018, 08:06

I wrote:

> We are just following up, are you still experiencing any issues? If so, please provide a screenshot and more detail of the steps you are taking so the support team is able to reproduce it. If you are not longer experiencing any issues, please let us know so we may close this ticket. Thank you and we look forward to your response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 13, 2018, 14:38

Kimberly wrote:

> It appears E2330 Briar Pipe is not linked to E13999 BlueTracs.  E2330 should be linked to this account, should I just add it or can you move the current one that is in SAB as E2330 to E13999?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 14, 2018, 03:27

Hi [Kyle] ( @kdjstudios ),  


By reading the comments so far we were hoping that she managed to update the account number of `123-E1330` to `123-E2330` by the way we mentioned on the [comment] (http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1145#note_27601) but by looking on the current account status the thing was not get done.

So, if you can please ask Kimberly that what exactly she wants now for us to do?   

And in case she just wants to link `E2330 Briar Pipe` to `E13999 BlueTracs` then please let us know.  


Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Nov 14, 2018, 05:59

@kdjstudios Kyle, I'm confused - can't they do the parent child links right in the UI?  Why would they need to submit a ticket about this?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 14, 2018, 06:00

Hi [Kyle] ( @kdjstudios ),   

We found a duplicate ticket for this #1129 and we have completed that today. So we think this is done.

Please let us know if there is something else to be done from our end.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 14, 2018, 12:07

@tim.traylor @pchaudhary @amishra @smasih 

Operations can do parent child accounts for accounts under the **same** customer. However, this request is for merging accounts under **different** customers. 

As it was just updated today in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1129#note_29777, now both accounts are under the same customer and are listed in a parent child account relation. 

I wrote:

> The support team has informed me that the account "123-E2330	Briar Pipe/Harbor Properties" is now listed as a child account to "123-E13999 BlueTracs". May you please confirm.

EDIT: To also clarify I created ticket #1126 about a month ago to start to discussion to why we removed the ability to switch customers through the GUI and what it will take to reestablish that functionality.
